cpp/src/gandiva/string_function_holder_test.cc

```diff
@@ -212,6 +213,189 @@ TEST_F(TestLikeHolder, TestMultipleEscapeChar) {
   auto status = LikeHolder::Make("ab\\_", "\\\\", &like_holder);
   EXPECT_EQ(status.ok(), false) << status.message();
 }
+
+// Tests related to the REGEXP_EXTRACT function
+class TestExtractHolder : public ::testing::Test {
+ protected:
+  ExecutionContext execution_context_;
+};
+
+TEST_F(TestExtractHolder, TestSimpleExtract) {
+  std::shared_ptr<ExtractHolder> extract_holder;
+
+  // Pattern to match of two group of letters
+  auto status = ExtractHolder::Make(R"((\w+) (\w+))", &extract_holder);
```

Comment:
Shouldn't this be R"(\w+) (\w+)" since you are enclosing another bracket inside the constructor?
So in your example index 2 should return first name?

Reply:
About the first question, the first and last parenthesis does not count, they are part of the [raw string sintax](https://www.geeksforgeeks.org/raw-string-literal-c/) in C++

Reply:
And in the example, the index 0 must return all string parts that match the regex, the index 1 the first group in the regex(first name) and the index 2 the second group(surname)